src/Symfony/Component/Messenger/Tests/Command/FailedMessagesRetryCommandTest.php

```diff
@@ -45,4 +49,101 @@ public function testBasicRun()
 
         $this->assertStringContainsString('[OK]', $tester->getDisplay());
     }
+
+    public function testBasicRunWithServiceLocator()
+    {
+        $receiver = $this->createMock(ListableReceiverInterface::class);
+        $receiver->expects($this->exactly(2))->method('find')->withConsecutive([10], [12])->willReturn(new Envelope(new \stdClass()));
+        // message will eventually be ack'ed in Worker
+        $receiver->expects($this->exactly(2))->method('ack');
+
+        $dispatcher = new EventDispatcher();
+        $bus = $this->createMock(MessageBusInterface::class);
+        // the bus should be called in the worker
+        $bus->expects($this->exactly(2))->method('dispatch')->willReturn(new Envelope(new \stdClass()));
+
+        $failureTransportName = 'failure_receiver';
+        $serviceLocator = $this->createMock(ServiceLocator::class);
+        $serviceLocator->method('has')->with($failureTransportName)->willReturn(true);
+        $serviceLocator->method('get')->with($failureTransportName)->willReturn($receiver);
+
+        $command = new FailedMessagesRetryCommand(
+            $failureTransportName,
+            $serviceLocator,
+            $bus,
+            $dispatcher
+        );
+
+        $tester = new CommandTester($command);
+        $tester->execute(['id' => [10, 12], '--force' => true]);
+
+        $this->assertStringContainsString('[OK]', $tester->getDisplay());
+        $this->assertStringNotContainsString('Available failure transports are:', $tester->getDisplay());
+    }
+
+    public function testBasicRunWithServiceLocatorMultipleFailedTransportsDefined()
+    {
+        $receiver = $this->createMock(ListableReceiverInterface::class);
+        $receiver->method('all')->willReturn([]);
+
+        $dispatcher = new EventDispatcher();
+        $bus = $this->createMock(MessageBusInterface::class);
+
+        $failureTransportName = 'failure_receiver';
+        $serviceLocator = $this->createMock(ServiceLocator::class);
+        $serviceLocator->method('has')->with($failureTransportName)->willReturn(true);
+        $serviceLocator->method('get')->with($failureTransportName)->willReturn($receiver);
+        $serviceLocator->method('getProvidedServices')->willReturn([
+            'failure_receiver' => [],
+            'failure_receiver_2' => [],
+            'failure_receiver_3' => [],
+        ]);
+
+        $command = new FailedMessagesRetryCommand(
+            $failureTransportName,
+            $serviceLocator,
+            $bus,
+            $dispatcher
+        );
+        $tester = new CommandTester($command);
+        $tester->setInputs([0]);
+        $tester->execute(['--force' => true]);
+
+        $expectedLadingMessage = <<<EOF
+> Loading messages from the global failure transport failure_receiver.
+> To use a different failure transport, pass --transport=.
+> Available failure transports are: failure_receiver, failure_receiver_2, failure_receiver_3
+EOF;
```

Comment:
I don't think we need to assert the entire, exact string. Probably the most important part is `failure transports are: failure_receiver, failure_receiver_2, failure_receiver_3`